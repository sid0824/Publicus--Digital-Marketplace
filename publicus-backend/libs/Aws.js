import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});