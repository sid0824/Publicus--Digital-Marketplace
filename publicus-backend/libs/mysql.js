import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};