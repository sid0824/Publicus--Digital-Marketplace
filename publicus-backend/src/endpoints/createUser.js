import {runQuery} from "../../libs/mysql";
import {failure, success} from "../util/ResponseUtil";

const defaultAvatar = 'http://s3.amazonaws.com/37assets/svn/765-default-avatar.png';

//Generate Insert SQL Query (adds new user to database)
const insertUserQuery = (userId, email, firstname, lastname) => {
  return `INSERT INTO Users (userId, email, avatar, firstname, lastname)
          VALUES ( '${userId}', '${email}', '${defaultAvatar}', '${firstname}', '${lastname}' )`;
};

export async function main(event, context) {
  try {
    //Retrieve body variables
    const data = JSON.parse(event.body);
    await runQuery(insertUserQuery(data.cognitoId, data.email, data.firstname, data.lastname));
    return success();
  } catch (err) {
    console.error(err);
    return failure();
  }
}
