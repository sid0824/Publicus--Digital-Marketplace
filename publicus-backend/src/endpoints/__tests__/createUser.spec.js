import { main } from '../createUser';
import { runQuery } from "../../../libs/mysql";

jest.mock("../../../libs/mysql");

const mockEvent = {
  requestContext: {
    identity: {
      cognitoIdentityId: 'MOCK_COGNITO_ID'
    },
  },
  body: JSON.stringify({
    cognitoId: 'COGNITO_ID',
    email: 'EMAIL',
    firstname: "FIRSTNAME",
    lastname: 'LASTNAME'
  }),
  pathParameters: {
    surveyId: 'MOCK_SURVEY_ID'
  }
};

describe('createUser', () => {

  beforeEach(() => {
    runQuery.mockResolvedValue(null);
  });

  afterEach(() => {
    runQuery.mockReset();
  });

  test(`should respond with failure when an error occurs`, async () => {
    runQuery.mockReset();
    runQuery.mockImplementation(() => {
      throw new Error('ERROR')
    });
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  });

  test(`should add a user to the database`, async () => {
    const response = await main(mockEvent);
    expect(runQuery.mock.calls[0][0]).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  })

});