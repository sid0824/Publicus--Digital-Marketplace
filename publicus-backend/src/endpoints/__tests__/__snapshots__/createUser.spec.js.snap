// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUser should add a user to the database 1`] = `
"INSERT INTO Users (userId, email, avatar, firstname, lastname)
          VALUES ( 'COGNITO_ID', 'EMAIL', 'http://s3.amazonaws.com/37assets/svn/765-default-avatar.png', 'FIRSTNAME', 'LASTNAME' )"
`;

exports[`createUser should add a user to the database 2`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 200,
}
`;

exports[`createUser should respond with failure when an error occurs 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;
